////////////////////////////////////////////////////////////////////////
// FBI Warning:
// Before you modify this file, you must ask related people if you can modify it.
// see https://developers.google.com/protocol-buffers/docs/proto
////////////////////////////////////////////////////////////////////////
syntax = "proto3";
package pb;

message LoginReq
{
    string user_name = 1;
    string password = 2;
    int32 fd = 3;
    int32 req_begin_time = 4;
}

message LoginResp
{
    string session_id = 1;
    int32 fd = 2;
    int32 req_begin_time = 3;
    int32 ret = 4;
    string ret_msg = 5;
}

message GnsInfoReq
{
    repeated string gns_name = 1;
    repeated string ip_port = 2;
}

message CfgInfoReq
{
    string session_id = 1;
    int32 p1 = 2;
    int64 p2 = 3;
    string p3 = 4;
    string gns_name = 5;
    string ip_port = 6;
    int32 fd = 7;
    int32 req_begin_time = 8;
}

message CfgInfoResp
{
    int32 ret= 1;
    string ret_msg = 2;
    int32 fd = 3;
    int32 req_begin_time = 4;
}

message LogInfo
{
    string gns_name = 1;
    string ip_port = 2;
    string session_id = 3;
    string log_text = 4;
}

message GetLogReq
{
    string session_id = 1;
    string gns_name = 2;
    string ip_port = 3;
    int32  seq = 4; 
    int32 fd = 5;
    int32 req_begin_time = 6;
}

message GetLogRespOld
{
    int32 ret = 1;
    string ret_msg = 2;
    string log_text = 3;
    int32  cur_len = 4; //current length of the gns corresponding vector
    int32 fd = 5;
    int32 req_begin_time = 6;
}

message LogText
{
    string ip_port = 1;
    string gns_name = 2;
    repeated string text = 3;
}

message GetLogResp
{
    int32 ret = 1;
    string ret_msg = 2;
    int32 fd = 3;
    int32 req_begin_time = 4;
    repeated LogText log_text = 5;
}

message HeartBeatMsg
{
    string reserve = 1;
}

message ClearSessionReq
{
    string session_id = 1;
}

